domains/client/domain-operator/src/domain_bundle_producer.rs

```diff
@@ -131,7 +139,16 @@ where
             // already processed a block higher than the local best and submitted the receipt to
             // the parent chain, we ought to catch up with the consensus block processing before
             // producing new bundle.
-            !domain_best_number.is_zero() && domain_best_number <= head_receipt_number
+            let is_operator_lagging =
+                !domain_best_number.is_zero() && domain_best_number <= head_receipt_number;
+
+            let skip_out_of_order_slot = self.skip_out_of_order_slot
```

Comment:
this feels like we should introduce conditionally compiled logic for tests if this is a specific behavior we need for domain tests.

This is the second bool that we introduced for the just the tests. I think there is a better approach

Reply:
Indeed, these bools are quite invasive to the production code. I'm thinking about introducing a trait for the 
 `DomainBundleProducer`, for the production code the actual `DomainBundleProducer` will be used while in the test a wrapper of `DomainBundleProducer` will be used, and both the `skip_empty_bundle_production` and `skip_out_of_order_slot` will be handled inside the wrapper without touching the production code. WDYT

Reply:
Since this is limited to just domain-operator tests, hence, i was leaning towards conditional compiled for tests so that we can get away from invasive change until we need it. 
But I don't mind introducing a wrapper right now as well 👍🏼 

We can get this in but do create an issue so that we can get rid of these bools. Also leave a TODO in this PR for future reference

Reply:
I'm leaning towards the wrapper approach since conditional compiled will still tough the production code and reduce readability.

Will leave a TODO in this PR and create an issue once this is merged.